packages/test-framework/src/fixture/pageObject/Checkout.ts

```diff
@@ -92,4 +92,152 @@ export class Checkout {
+        await this.page.locator(`${formId} [data-test="phoneInput-text"]`).fill(address.phone);
+        await this.page
+            .locator(`${formId} [data-test="addressLine1Input-text"]`)
+            .fill(address.address1);
+        await this.page
+            .locator(`${formId} [data-test="addressLine2Input-text"]`)
+            .fill(address.address2);
+        await this.page.locator(`${formId} [data-test="cityInput-text"]`).fill(address.city);
+        await this.page
+            .locator(`${formId} [data-test="countryCodeInput-select"]`)
+            .selectOption(address.countryCode);
+        await this.page
+            .locator(`${formId} [data-test="provinceCodeInput-select"]`)
+            .selectOption(address.stateOrProvinceCode);
+        await this.page
+            .locator(`${formId} [data-test="postCodeInput-text"]`)
+            .fill(address.postalCode);
+    }
+
+    async completeShippingAddressForm(
+        isBillingAddressSame = true,
+        shouldSaveAddress = true,
+        addressAppendText?: string,
+    ): Promise<void> {
+        await this.page
+            .locator('[data-test="shipping-address-heading"]')
+            .waitFor({ state: 'visible' });
+
+        await this.fillAddressForm({ formId: '#checkoutShippingAddress', addressAppendText });
+
+        if (!isBillingAddressSame) await this.page.locator('label[for="sameAsBilling"]').click();
+        if (!shouldSaveAddress)
+            await this.page.locator('label[for="shippingAddress.shouldSaveAddress"]').click();
+
+        await this.page.locator('#checkout-shipping-options').waitFor({ state: 'visible' });
+
+        const firstShippingMethod = this.page
+            .locator('#checkout-shipping-options .shippingOptions-container li')
+            .nth(0);
+
+        await firstShippingMethod.click();
+        await this.page.locator('#checkout-shipping-continue').click();
+    }
+
+    async completeBillingAddressForm(): Promise<void> {
+        await this.page
+            .locator('[data-test="billing-address-heading"]')
+            .waitFor({ state: 'visible' });
+
+        await this.fillAddressForm({ formId: '#checkoutBillingAddress' });
+
+        await this.page.locator('#checkout-billing-continue').click();
+    }
+
+    async fillNewAddressPopup(addressAppendText: string): Promise<void> {
+        await this.page.locator('.dropdownMenu').waitFor({ state: 'visible' });
+        await this.page.locator('.dropdownMenu li').nth(0).click();
+        await this.page
+            .locator('.ReactModalPortal .modal.optimizedCheckout-contentPrimary')
+            .waitFor({ state: 'visible' });
+        await this.fillAddressForm({
+            formId: '[data-test="modal-body"] .checkout-address',
+            addressAppendText,
+        });
+        await this.page.locator('#checkout-save-address').click();
+    }
+
+    async completeMultiShippingAddressForm(): Promise<void> {
```

Comment:
A shopper can also finish the multishipping form by selecting an existing address?
Can we cover that use case? 
Meanwhile, perhaps we can rename this method as `completeMultiShippingWithTwoNewAddresses`.

Reply:
I changed existing test case's second item to have select address from saved address.